electron/src/sso/SingleSignOn.ts

```diff
@@ -213,16 +213,19 @@ export class SingleSignOn {
     this.session.setPermissionRequestHandler((webContents, permission, callback) => callback(false));
 
     // User-agent normalization
-    this.session.webRequest.onBeforeSendHeaders({urls: ['*']}, (details: any, callback: Function) => {
-      details.requestHeaders['User-Agent'] = config.userAgent;
+    this.session.webRequest.onBeforeSendHeaders({urls: ['*']}, (details, callback) => {
+      details.requestHeaders = {
```

Comment:
Do you really need to put it back into the `details` object? (I mean mutate this object).
Since it's given to the callback, it might not be needed. WDYT?

Reply:
The problem is, when I write it as 
```ts
details.requestHeaders['User-Agent'] = config.userAgent;
```
TypeScript won't allow it since the `RequestHeaders` in Electron is empty:

```ts
interface RequestHeaders {
}
```
and it doesn't expect `'User-Agent'` in there:

```
Element implicitly has an 'any' type because expression of type '"User-Agent"' can't be used to index type 'RequestHeaders'.
Property 'User-Agent' does not exist on type 'RequestHeaders'. ts(7053)
```

Anyway, I found out that Electron is using `any` there, too:
https://github.com/electron/electron/blob/cfd230d7f1905973113ded0b0b9965dbc4ad46b4/spec/ts-smoke/electron/main.ts#L1173

:roll_eyes:

Reply:
Oh no, what I meant was:

```
const headers = { /*...*/}

callback({ ..., requestHeaders: headers });
```

instead of mutation the `details` object (but maybe Electron relies on this mutation, that would be ugly but, well, ...

Reply:
I'm going to use `any` for now.